index.js

```diff
@@ -115,77 +115,251 @@ function initCarousel(options) {
+const modalimg = modalventana.querySelector ('.js-ventana-modal-img');
+
+let currentIndex = 0;
+
+function changeInterfaceColors(dishInfo) {
+  const colorSet = {
+    ellipseColor: dishInfo.getAttribute('ellipseColor'),
+    shadowColor: dishInfo.getAttribute('shadowColor'),
+    buttonColor: dishInfo.getAttribute('buttonColor'),
+    hoverColor: dishInfo.getAttribute('hoverColor'),
+  };
+
+  modalventana.style.setProperty('--modal-color', colorSet.buttonColor);
+  modalbutton.style.setProperty('--hover-modal', colorSet.hoverColor);
+  modalbutton.style.setProperty('--shadow-modal', colorSet.shadowColor);
+  modalcontent.style.setProperty('--modal-box', colorSet.shadowColor);
+  modalclose.style.setProperty('--modal-boxx', colorSet.shadowColor);
+
+  ellipse.style.setProperty('--bg-color', colorSet.ellipseColor);
+  pagination.style.setProperty('--bg-orange', colorSet.buttonColor);
+  price.style.setProperty('--bg-orange', colorSet.buttonColor);
+  colorsvg.style.setProperty('--bg-orange', colorSet.buttonColor);
+
+  prevBtn.style.setProperty('--bg-color', colorSet.buttonColor);
+  prevBtn.style.setProperty('--shadow-bg', colorSet.shadowColor);
+  prevBtn.style.setProperty('--hover-color', colorSet.hoverColor);
+
+  nextBtn.style.setProperty('--bg-color', colorSet.buttonColor);
+  nextBtn.style.setProperty('--shadow-bg', colorSet.shadowColor);
+  nextBtn.style.setProperty('--hover-color', colorSet.hoverColor);
+
+  button.style.setProperty('--bg-orange', colorSet.buttonColor);
+  button.style.setProperty('--shadow-bg', colorSet.shadowColor);
+  button.style.setProperty('--hover-color', colorSet.hoverColor);
+
+  menuItems.forEach(item => {
+    item.style.setProperty('--hover-color', colorSet.hoverColor);
+  });
+
+  ItenSubmenu.forEach(submenu => {
+    submenu.style.setProperty('--hover-color', colorSet.hoverColor);
+  });
+
+  price.style.color = colorSet.buttonColor;
+}
+
+function updateDishDetails(currentDish) {
+  const dishInfo = currentDish.querySelector('dishe-plato');
+  changeInterfaceColors(dishInfo);
+  image = dishInfo.getAttribute('image');
+  updatemodalimg(image);
+  updatePrice(dishInfo.getAttribute('price'));
+  updatemodalprice(dishInfo.getAttribute('price'));
+  updatemodalnumber(dishInfo.getAttribute('discount'));
+  updateTitle(dishInfo.getAttribute('title'));
+  updatemodaltitle(dishInfo.getAttribute('title'));
+  updateDescription(dishInfo.getAttribute('description'));
+  updatemodalparagraph(dishInfo.getAttribute('description'));
+}
+
+nextBtn.addEventListener('click', () => {
+  currentIndex = (currentIndex + 1) % dishesPlatos.length; 
+  updateDishDetails(dishesPlatos[currentIndex]);
+});
+
+prevBtn.addEventListener('click', () => {
+  currentIndex = (currentIndex - 1 + dishesPlatos.length) % dishesPlatos.length; 
+  updateDishDetails(dishesPlatos[currentIndex]);
+});
 
-let currentPrice = 32;
+function updatemodalimg(image) {
+  modalimg.src = image; 
```

Comment:
Como ese objeto `modalimg` solo se usa aqui, deberia estar definido aqui, no donde lo colocaste
```suggestion
  const modalimg = modalventana.querySelector ('.js-ventana-modal-img');
  modalimg.src = imageSource;
```